applications: serial_lte_modem: Store credentials to settings (#13610)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/SOCKET_AT_commands.rst

* The ``<sec_tag>`` parameter is an integer.
  It indicates to the modem the credential of the security tag to be used for establishing a secure connection.
  It is associated with a credential, that is, a certificate or PSK. The credential should be stored on the modem side beforehand.

>   Note: ``<sec_tag>`` can only be used with ``<role>`` ``1`` - Server, when ``<type>`` is ``SOCK_STREAM`` and the overlay-native_tls.conf configuration file is used.

Author: This actually changes the meaning. You can use sec_tag when the role has value 0 (client), but with role 1 (server), you can only use it if type has the value of SOCK_STREAM (this requirement will vanish in my next PR) and the overlay is used.

>   Note: ``<sec_tag>`` can only be used with ``<role>`` ``1`` - Server, when ``<type>`` is ``SOCK_STREAM`` and the overlay-native_tls.conf configuration file is used.

Reviewer: Ah, now I see the meaning. Then:
```suggestion
  Note that when ``<role>`` has a value of ``1``, ``<sec_tag>`` can only be used if ``<type>`` has a value of ``SOCK_STREAM`` and the :file:`overlay-native_tls.conf` configuration file is used.
```
